build-aux/snap/snapcraft.yaml

```diff
@@ -24,15 +24,17 @@ parts:
   foliate:
     plugin: meson
     source: .
+    override-pull: |
+      snapcraftctl pull
+      sed -i -e 's|@GJS@|/usr/bin/gjs|g' src/com.github.johnfactotum.Foliate.in
+      snapcraftctl set-version $(git describe --tags --abbrev=10)
```

Comment:
In the current version of the snap, we use the AppStream file to get the version. That works great for releases, but not for the daily builds in edge; they just get the same version as the previous stable release. I think it would be cleaner to have Meson generate the version in the AppStream file using git describe. Then we kan keep using the version from AppStream and all development builds will have a useful version string, instead of only the snap. However, I don't know if there is a good way to do that in Meson.

@johnfactotum What do you think about this? Should we try to get Meson to generate the version, or do it in the snap here?

Reply:
I'm not really sure. In the past I've appended the commit hash to the version when built with the development option, and also added the `.devel` class, which, in Adwaita at least, adds a blue tint and cogs to the headerbar background (see #118). However I've not done this in the 2.0 rewrite, because it doesn't seem very useful.

In many GNOME apps, they have an option to build the development version separate from the stable version, and it's possible to install them side-by-side. Their development builds also use modified icons.

I think the proper way is to have Meson generate everything properly. Another option is to manually bump the version and add a development release to AppData after each stable release. This could be applied on its own or in conjunction with Meson adding the git hash.

But all these are pretty cumbersome to implement and personally I don't see a lot of advantage.

Reply:
Got it!